src/harness/tests/cbsd_sas_deregistration_test.py

```diff
@@ -0,0 +1,260 @@
+        self.assertEqual(response[0]['response']['responseCode'], 102)
+        self.assertNotIn('cbsdId', regResponse[0])
+
+    def test_10_15_4_2_2(self):
+        self._sas.logger.info(
+            '10.15.4.2.2 Missing CBSD ID (TBD) : '
+            'two objects in the DeregistrationRequest')
+
+        device1_a = json.load(
+            open(os.path.join(
+                os.getcwd(),
+                'tests', 'testdata', 'registration', 'reg1_a.json')))
+
+        device2_a = json.load(
+            open(os.path.join(
+                os.getcwd(),
+                'tests', 'testdata', 'registration', 'reg2_a.json')))
+
+        # Register
+        devices = [device1_a, device2_a]
+        response = self._sas.Registration(devices)
+        self.assertEqual(response[0]['response']['responseCode'], 0)
+        self.assertEqual(response[1]['response']['responseCode'], 0)
+
+        # Deregistration
+        regResponse = response
+        del regResponse[0]['cbsdId']
+        response = self._sas.Deregistration(regResponse)
+        self.assertEqual(response[0]['response']['responseCode'], 0)
+        self.assertEqual(response[0]['cbsdId'], regResponse[0]['cbsdId'])
+        self.assertEqual(response[1]['response']['responseCode'], 102)
+        self.assertNotIn('cbsdId', regResponse[1])
+
+    def test_10_15_4_3_1(self):
+        self._sas.logger.info(
+            '10.15.4.3.1 CBSD ID Does Not Exist in the SAS : '
+            'single request object')
+
+        devices = json.load(
+            open(os.path.join(
+                os.getcwd(),
+                'tests', 'testdata', 'registration', 'reg1_a.json')))
+
+        # Register
+        response = self._sas.Registration(devices)
+        self.assertEqual(response[0]['response']['responseCode'], 0)
+
+        # Deregistration
+        regResponse = response
+        regResponse[0]['cbsdId'] = 'dummy'
+        response = self._sas.Deregistration(regResponse)
+        self.assertEqual(response[0]['response']['responseCode'], 103)
+
+    def test_10_15_4_3_2(self):
+        self._sas.logger.info(
+            '10.15.4.3.2 CBSD ID Does Not Exist in the SAS : '
+            'two request objects')
+
+        device1_a = json.load(
+            open(os.path.join(
+                os.getcwd(),
+                'tests', 'testdata', 'registration', 'reg1_a.json')))
+
+        device2_a = json.load(
+            open(os.path.join(
+                os.getcwd(),
+                'tests', 'testdata', 'registration', 'reg2_a.json')))
+
+        # Register
+        devices = [device1_a, device2_a]
+        response = self._sas.Registration(devices)
+        self.assertEqual(response[0]['response']['responseCode'], 0)
+        self.assertEqual(response[1]['response']['responseCode'], 0)
+
+        # Deregistration
+        regResponse = response
+        regResponse[1]['cbsdId'] = 'dummy'
+        response = self._sas.Deregistration(regResponse)
```

Comment:
but if I understand your code correctly you will also include 'responseCode' field which should not be in request.